api/README.md

```diff
@@ -1,6 +1,6 @@
 OpenTelemetry API
 ======================================================
 
-* Java 7 and Android 14 compatible.
```

Comment:
API is still level 14

Reply:
so perhaps its not at 24 yet like @jkwatson mentioned?  So it should read "Android 4.0 (API level 14)"

Reply:
Yes, the api and context_prop artifacts are still at "Android 4.0 (API level 14)"

Reply:
Updated.  @bogdandrutu I had someone ask me about android 14 and they requested clarification if API level or android version. (helping out where i can)